crates/orderbook/src/cow_subsidy.rs

```diff
@@ -100,16 +105,21 @@ impl CowSubsidyImpl {
         );
 
         Self {
-            cow_token,
+            token,
+            vtoken,
             subsidy_tiers,
             cache: Mutex::new(cache),
         }
     }
 
     async fn subsidy_factor_uncached(&self, user: H160) -> Result<f64> {
-        let balance = self.cow_token.balance_of(user).call().await?;
-        let tier = self.subsidy_tiers.0.range(..=balance).rev().next();
-        Ok(tier.map(|tier| *tier.1).unwrap_or(1.0))
+        let balance = self.token.balance_of(user).call().await?;
+        let vbalance = self.vtoken.swappable_balance_of(user).call().await?;
```

Comment:
I think those balances could get fetched in parallel.

Reply:
Maybe try out the `BufferedTransport` so that you can `join` the futures and get batched calls for free? 😄.

Reply:
Hmm, I think this is wrong. I think that the total subsidy should be:
```suggestion
        let balance = self.token.balance_of(user).call().await?;
        let vbalance = self.vtoken.balance_of(user).call().await?;
```

Let me double check.

Reply:
Right, it is the combined balance and not just the swappable vCOW balance. From the CIP:

> Combined balance of vCOW and COW is accounted for per network 

https://forum.cow.fi/t/cip-3-trading-fee-discount-for-cow-holders/325

Reply:
So swappable balance of is correct or balance of? The docs https://github.com/gnosis/cow-token/blob/main/src/contracts/CowProtocolVirtualToken.sol say that balance_of would include future vested amounts which seemed wrong.

Reply:
I personally read the quoted section such that it's supposed to highlight the fact that only the balance of the current chain is considered while being a little vague on the exact calculation of the "correct" balance.
IMO opinion not considering the unvested amounts makes more sense, but depending on how you read the CIP it would be wrong.

Reply:
> IMO opinion not considering the unvested amounts makes more sense

Interesting. I would argue the opposite. Unvested vCOW holders are locked in for 4 years and are way more long on COW than regular holders - so to me it makes sense for them to also get benefits.

I asked on Slack to confirm.

Reply:
I was also expecting it to be the cow balance plus the still vested balance. Not just the cow plus swappable vCOW balance.

Reply:
It was confirmed by Anna on Slack that we want the full vCOW token balance and not just the vested (a.k.a. swappable) balance for the purpose of computing subsidies.

Reply:
@vkgnosis - sorry but I high jacked this comment and I think the original suggestion by @MartinquaXD was missed:

> I think those balances could get fetched in parallel.